bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -81,20 +102,97 @@
 
   private final String host;
 
+  private final HeapSizeManager heapSizeManager;
   private final AsyncExecutor asyncExecutor;
+  private final ExecutorService executorService;
+  private final BigtableOptions options;
+
+  private final LinkedBlockingQueue<MutationOperation> mutationsToBeSent =
+      new LinkedBlockingQueue<>();
+
+  /**
+   * A counter for the number of {@link #mutationWorker}s are
+   * active.
+   */
+  private final AtomicInteger activeMutationWorkers = new AtomicInteger();
+
+  /**
+   * This {@link Runnable} pulls a mutation from {@link #mutationsToBeSent}, and calls {{@link
+   * #issueRequest(Mutation, long)}.
+   */
+  private final Runnable mutationWorker = new Runnable() {
+    @Override
+    public void run() {
+      activeMutationWorkers.incrementAndGet();
+      try {
+        while (!closed) {
+          try {
+            MutationOperation operation =
+                mutationsToBeSent.poll(MUTATION_TO_BE_SENT_WAIT_MS, TimeUnit.MILLISECONDS);
+            // The operation can be null if a timeout occurs.
+            if (operation != null) {
```

Comment:
From BlockingQueue javadoc: 

``A BlockingQueue does not intrinsically support any kind of "close" or "shutdown" operation to indicate that no more items will be added. The needs and usage of such features tend to be implementation-dependent. For example, a common tactic is for producers to insert special end-of-stream or poison objects, that are interpreted accordingly when taken by consumers. ```

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html